internal/resources/managementclusterregistration/datasource_management_cluster_registration.go

```diff
@@ -0,0 +1,69 @@
+/*
+Copyright © 2023 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package managementclusterregistration
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pkg/errors"
+
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/helper"
+	managementclusterregistrationmodel "github.com/vmware/terraform-provider-tanzu-mission-control/internal/models/managementclusterregistration"
+	"github.com/vmware/terraform-provider-tanzu-mission-control/internal/resources/common"
+)
+
+func DataSourceManagementClusterRegistration() *schema.Resource {
+	return &schema.Resource{
+		ReadContext: func(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+			return dataSourceClusterRead(helper.GetContextWithCaller(ctx, helper.DataRead), d, m)
+		},
+		Schema: managementClusterRegistrationSchema,
+	}
+}
+
+func dataSourceClusterRead(_ context.Context, d *schema.ResourceData, _ interface{}) diag.Diagnostics {
+	var (
+		diags diag.Diagnostics
+		resp  *managementclusterregistrationmodel.VmwareTanzuManageV1alpha1ManagementclusterCreateManagementClusterResponse
+		err   error
+	)
+
+	// TODO - get cluster resource
```

Comment:
TODO will be addressed in the follow up PR?

Reply:
Yes.